database-commons/src/main/java/io/cdap/plugin/util/RetryPolicyUtil.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.plugin.util;
+
+import dev.failsafe.Failsafe;
+import dev.failsafe.FailsafeException;
+import dev.failsafe.RetryPolicy;
+import io.cdap.cdap.api.Config;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.plugin.db.RetryExceptions;
+import io.cdap.plugin.db.connector.AbstractDBConnectorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.SQLTransientException;
+import java.time.Duration;
+
+
+/**
+ * Utility class for creating standardized {@link dev.failsafe.RetryPolicy} configurations
+ * to handle transient SQL exceptions using the Failsafe library.
+ *
+ * <p>This class centralizes the retry logic parameters like backoff delay, max retry count,
+ * and retry duration, and provides logging support for retry and failure events.</p>
+ */
+public class RetryPolicyUtil extends Config {
+    public static final Logger LOG = LoggerFactory.getLogger(RetryPolicyUtil.class);
+    private static AbstractDBConnectorConfig config;
+    public static final RetryPolicy<Object> RETRY_POLICY = RetryPolicyUtil.createConnectionRetryPolicy(config);
```

Comment:
Is it necessary to store the return value of `createConnectionRetryPolicy(config)` method in the `RETRY_POLICY` variable? Can't we directly call this static method wherever required?

Reply:
`config` will always be `null` Pass the config as an argument when directly calling the method `createConnectionRetryPolicy(config)`